lib/utils/message.js

```diff
@@ -0,0 +1,95 @@
+/*!
+ * message.js - message signing utilities.
+ * Copyright (c) 2019, The Bcoin Developers (MIT License).
+ */
+
+'use strict';
+
+const assert = require('bsert');
+const bufio = require('bufio');
+const hash256 = require('bcrypto/lib/hash256');
+const secp256k1 = require('bcrypto/lib/secp256k1');
+
+/**
+ * @exports utils/message
+ */
+
+const message = exports;
+
+/**
+ * Bitcoin signing magic string.
+ * @const {String}
+ * @default
+ */
+
+message.MAGIC_STRING = 'Bitcoin Signed Message:\n';
+
+/**
+ * Hash message with magic string.
+ * @param {String} message
+ * @param {String} [prefix = message.MAGIC_STRING]
+ * @returns {Hash}
+ */
+
+message.magicHash = (msg, prefix = message.MAGIC_STRING) => {
+  assert(typeof prefix === 'string', 'prefix must be a string.');
+  assert(typeof msg === 'string', 'message must be a string');
+
+  const bw = bufio.write();
+
+  bw.writeVarString(prefix);
+  bw.writeVarString(msg, 'utf8');
+
+  return hash256.digest(bw.render());
+};
+
+/**
+ * Sign message with key.
+ * @param {String} msg
+ * @param {KeyRing} ring
+ * @param {String} [prefix = message.MAGIC_STRING]
+ * @returns {Buffer}
+ */
+
+message.sign = (msg, ring, prefix) => {
+  assert(ring.getPrivateKey(), 'Cannot sign without private key.');
+
+  const hash = message.magicHash(msg, prefix);
+  const compress = 0x04 !== ring.getPublicKey().readInt8(0);
+  const {
+    signature,
+    recovery
+  } = secp256k1.signRecoverable(hash, ring.getPrivateKey());
```

Comment:
I'm getting `Error: secp256k1.signRecoverable is not a function` running tests locally. Looks like this method was added in a somewhat recent commit:

https://github.com/bcoin-org/bcrypto/commit/e553ba1ed4b8d02b28f0bfbd6251f140dd5bd6cf

Should this PR also include a `package.json` bump to `4.0.1` ?